### ISSUE TITLE
Add support for network aliases

### ISSUE THREAD

**Issue by jadkik** [containers/ansible-podman-collections#308]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

None of the modules in this collection have support for network aliases. It seems like a feature that was (relatively) recently added to podman: https://github.com/containers/podman/issues/4530#issuecomment-724935712

It would be cool to support this in one of those ways:

- Support for the `podman network connect --alias` command [documented here](https://docs.podman.io/en/latest/markdown/podman-network-connect.1.html#alias)
- Support for the `podman container create --network-alias` command [documented here](https://docs.podman.io/en/latest/markdown/podman-create.1.html#network-alias-alias)

I could try to submit a PR but I am not sure how idempotency would work in this case (e.g. `podman network connect --alias X` would fail when a new alias `X` is added - if the container is already in the network)

**Additional information you deem important (e.g. issue happens only occasionally):**

The workaround I am using so far is similar to this:

```yaml
- name: Workaround to add alias to memcached container
  shell: |-
    podman network disconnect network-name container-name || true
    podman network connect --alias alias-name network-name container-name
```

**Version of the `containers.podman` collection:**
**Either git commit if installed from git: `git show --summary`**
**Or version from `ansible-galaxy` if installed from galaxy: `ansible-galaxy collection list | grep containers.podman`**

```
1.8.0 (the command above does not work with my Ansible version)
```

**Output of `ansible --version`:**

```
ansible 2.9.12
  config file = /Users/jadkik/code/personal/tlakua/ansible.cfg
  configured module search path = ['/Users/jadkik/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jadkik/.pyenv/versions/3.8.1/envs/venv-p/lib/python3.8/site-packages/ansible
  executable location = /Users/jadkik/.pyenv/versions/venv-p/bin/ansible
  python version = 3.8.1 (default, Feb 20 2020, 11:02:42) [Clang 11.0.0 (clang-1100.0.33.17)]
```

**Output of `podman version`:**

```
Version:      3.2.3
API Version:  3.2.3
Go Version:   go1.15.14
Built:        Tue Aug 10 22:55:16 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.3-0.10.module_el8.4.0+886+c9a8d9ad.x86_64
```

**Playbok you run with ansible (e.g. content of `playbook.yaml`):**

```
(paste your output here)
```

**Command line and output of ansible run with high verbosity**

**Please NOTE: if you submit a bug about idempotency, run the playbook with `--diff` option, like:**

`ansible-playbook -i inventory --diff -vv playbook.yml`

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by sshnaidm** [comment]
That looks interesting. Would you like to make a patch? :)

**Comment by jadkik** [comment]
I'll try to make a patch :) Not sure how to make it idempotent, run in check mode, etc.

Will mention it here when the patch is ready. I'll be doing this in my free time so it might take some time.

**Comment by sshnaidm** [comment]
> I'll try to make a patch :) Not sure how to make it idempotent, run in check mode, etc.
> 
> Will mention it here when the patch is ready. I'll be doing this in my free time so it might take some time.

@jadkik let's start from non-idempotent and without check mode, that's fine to add it later.

**Comment by jadkik** [comment]
@sshnaidm I got this draft PR: https://github.com/containers/ansible-podman-collections/pull/314

It works on my own machine but I didn't manage to run the tests locally and CI won't run without approvals. Not sure if I understand how the testing setup works.

Also I only added a test for the "rootful" network scenario because it's the only one which seems to create a network separately before creating/starting the container. The rootless one uses `container:X` notation which I'm not familiar with; I can modify that once I figure out how to test the first part.
